### PR TITLE
Upgrade to spark 1.5.0

### PR DESCRIPTION
**Pull request by koertkuipers**
The strangest thing was that i had to modify the ssl settings and add a dummy keystore. Without this change the test would fail with error about keyStore not being found. I did not have this issue before.

### CODE REVIEW

File: spark-mongodb/src/main/scala/com/stratio/provider/mongodb/reader/MongodbReader.scala [hunk 1]
```diff
@@ -125,7 +125,7 @@ class MongodbReader(
   }
 
   private def convertToStandardType(value: Any): Any = {
-    if (value.isInstanceOf[UTF8String])
+    if (value.isInstanceOf[StringType])
```
Comment (koertkuipers) [inline spark-mongodb/src/main/scala/com/stratio/provider/mongodb/reader/MongodbReader.scala]: not sure if this should be StringType or org.apache.spark.unsafe.types.UTF8String

**Comment by kanielc** [comment]
I made similar changes in my local branch.  There is a problem though in that this PR would mean that Spark 1.5.0 would be required for all users, since that function ``` findTightestCommonTypeOfTwo ``` doesn't exist in earlier versions of Spark.

**Comment by koertkuipers** [comment]
doesn't the relocation of UTF8String cause similar issues? i guess you can use class name comparison in that case..

given how much they keep changing the developer api i am not sure i would even try to make one version that runs on spark 1.3 - 1.5

**Comment by koertkuipers** [comment]
since you already made these changes i will close this

**Comment by kanielc** [comment]
My changes are local.  Your changes are more comprehensive.  I'd prefer your branch to my own.  You should re-open.

I think they'll have to increase the version and make it clear that this new version only would work with Spark 1.5+ (or create a split release, but that's additional complexity too).

**Comment by ccaballe** [comment]
Hi,

We have created a new branch feature/spark-1.5.0 for this purpose. You can do a new PR in that branch.

We plan to do some new features for a new  release with spark-1.4.X and we are thinking how to make a release compatible with 1.5 and 1.4

Thanks to all.

**Comment by koertkuipers** [comment]
This has been superseded by https://github.com/Stratio/spark-mongodb/pull/35

**Pull request closed without merging**